newIDE/app/src/Utils/LocalFileDownloader.spec.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import { sanitizeUrlWithEncodedPath } from './LocalFileDownloader';
+
+describe('LocalFileDownloader', () => {
+  describe('sanitizeUrlWith', () => {
+    test('sanitize filenames', () => {
+      expect(
+        sanitizeUrlWithEncodedPath(
+          'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-ROOM1.jpg'
+        )
+      ).toBe(
+        'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-ROOM1.jpg'
+      );
+      expect(
+        sanitizeUrlWithEncodedPath(
+          'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-testé.jpg'
+        )
+      ).toBe(
+        'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-test%C3%A9.jpg'
+      );
+      expect(
+        sanitizeUrlWithEncodedPath(
+          'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-testé 汉字.jpg'
+        )
+      ).toBe(
+        'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-test%C3%A9%20%E6%B1%89%E5%AD%97.jpg'
+      );
+      expect(
+        sanitizeUrlWithEncodedPath(
+          'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-testé 👋.jpg'
+        )
+      ).toBe(
+        'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-test%C3%A9%20%F0%9F%91%8B.jpg'
+      );
+      expect(
+        sanitizeUrlWithEncodedPath(
+          'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-testé 👋.jpg?token=abc'
+        )
+      ).toBe(
+        'https://project-resources.gdevelop.io/1234-56a1-4cf9-8e84-123456789/resources/123456789-test%C3%A9%20%F0%9F%91%8B.jpg%3Ftoken%3Dabc'
```

Comment:
Out of caution, can you add a check for this: 
<img width="989" alt="image" src="https://github.com/4ian/GDevelop/assets/1280130/ef99273f-2514-4794-98cb-55479d18ef81">

https://en.wikipedia.org/wiki/URL